flow-decomposition/src/main/java/com/powsybl/flow_decomposition/FlowDecompositionComputer.java

```diff
@@ -162,94 +149,64 @@ private LoadFlowRunningService.Result runDcLoadFlow(Network network) {
         NodalInjectionComputer nodalInjectionComputer = new NodalInjectionComputer(networkMatrixIndexes);
-        Map<String, Double> dcNodalInjection = getDcNodalInjection(flowDecompositionResults, networkMatrixIndexes);
+        Map<String, Double> dcNodalInjection = getDcNodalInjection(networkMatrixIndexes);
 
-        return getNodalInjectionsMatrix(network, flowDecompositionResults, netPositions, glsks,
+        return getNodalInjectionsMatrix(network, netPositions, glsks,
             nodalInjectionComputer, dcNodalInjection);
     }
 
-    private Map<String, Double> getDcNodalInjection(FlowDecompositionResults flowDecompositionResults,
-                                                    NetworkMatrixIndexes networkMatrixIndexes) {
+    private Map<String, Double> getDcNodalInjection(NetworkMatrixIndexes networkMatrixIndexes) {
         ReferenceNodalInjectionComputer referenceNodalInjectionComputer = new ReferenceNodalInjectionComputer(networkMatrixIndexes);
-        Map<String, Double> dcNodalInjection = referenceNodalInjectionComputer.run();
-        flowDecompositionResults.saveDcNodalInjections(dcNodalInjection);
-        return dcNodalInjection;
+        return referenceNodalInjectionComputer.run();
     }
 
     private SparseMatrixWithIndexesTriplet getNodalInjectionsMatrix(Network network,
-                                                                    FlowDecompositionResults flowDecompositionResults,
                                                                     Map<Country, Double> netPositions,
                                                                     Map<Country, Map<String, Double>> glsks,
                                                                     NodalInjectionComputer nodalInjectionComputer,
                                                                     Map<String, Double> dcNodalInjection) {
-        SparseMatrixWithIndexesTriplet nodalInjectionsMatrix =
-            nodalInjectionComputer.run(network,
-                glsks, netPositions, dcNodalInjection);
-        flowDecompositionResults.saveNodalInjectionsMatrix(nodalInjectionsMatrix);
-        return nodalInjectionsMatrix;
+        return nodalInjectionComputer.run(network, glsks, netPositions, dcNodalInjection);
     }
 
-    private void saveDcReferenceFlow(FlowDecompositionResults flowDecompositionResults, List<Branch> xnecList) {
-        flowDecompositionResults.saveDcReferenceFlow(getXnecReferenceFlows(xnecList));
+    private void saveDcReferenceFlow(FlowDecompositionResultsBuilder flowDecompositionResultsBuilder, List<Branch> xnecList) {
+        flowDecompositionResultsBuilder.saveDcReferenceFlow(getXnecReferenceFlows(xnecList));
     }
 
-    private SensitivityAnalyser getSensitivityAnalyser(Network network, NetworkMatrixIndexes networkMatrixIndexes) {
+    SensitivityAnalyser getSensitivityAnalyser(Network network, NetworkMatrixIndexes networkMatrixIndexes) {
```

Comment:
I understand that you need to make it private package to allow testing, but can't you find another solution that avoid breaking encapsulation ?